File: tests/code_index/fixtures/cpp/service.cpp
#include <string>

namespace demo {

class Service {
public:
    static const int MAX = 3;
    Service(int n) : count(n) {}
    int helper(int x) const { return x * 2; }
    int process(int v);
private:
    int count;
};

int Service::process(int v) {
    return helper(v + count);
}

template <typename T>
T identity(T x) { return x; }

}  // namespace demo

int free_function(int v) {
    demo::Service s(v);
    return s.process(v);
}

File: tests/code_index/fixtures/java/Service.java
package demo;

public class Service {
    public static final int MAX_RETRIES = 3;
    private int counter = 0;

    public Service(int n) {
        this.counter = n;
    }

    public static int helper(int x) {
        return x * 2;
    }

    public int process(String s) {
        return helper(s.length() + counter);
    }
}

interface Greeter {
    String hello();
}

enum Color {
    RED, GREEN, BLUE
}
